test/mux.sh

```diff
@@ -237,15 +243,26 @@ write_and_verify_logs() {
 }
 
 reset_ES_HOST() {
-    oc set env dc logging-mux $1 $2
+    muxpod=$( get_running_pod mux )
+    os::cmd::try_until_success "oc set env dc logging-mux $1 $2"
     os::cmd::try_until_failure "oc get pod $muxpod"
+    os::log::debug $( oc get pods -l component=mux )
     os::cmd::try_until_text "oc get pods -l component=mux" "^logging-mux-.* Running "
-    muxpod=$( get_running_pod mux )
 }
 
 cleanup() {
     local return_code="$?"
     set +e
+
+    # In case test failed in Test case FILE_BUFFER_STORAGE_TYPE: $MUX_FILE_BUFFER_STORAGE_TYPE 
+    # reset ES_HOST and OPS_HOST
+    reset_ES_HOST $ES_HOST_BAK $OPS_HOST_BAK
+
+    # wait for the file buffer disappears once
+    os::cmd::try_until_failure 'oc get pods | egrep "logging-mux-.*-deploy"'
+    os::cmd::try_until_failure 'oc get pods | egrep "logging-mux-.*Terminating"'
```

Comment:
These two lines do not seem right to me - what are you trying to do?  Follow a deployment which you have deployed either explicitly using `oc rollout` or implicitly by editing the mux dc, `oc set env dc/logging-mux`, or etc.?  If so, you should use `oc rollout status -w dc/logging-mux` which will follow the deployment and either return success when it is deployed, or failure if the deployment failed or timed out.

Reply:
Or
If `$muxpod` is set, can't you just do 
```
os::cmd::try_until_failure "oc get pod $muxpod"
```
?

Reply:
Well, I followed the test steps manually and saw these lines in the transition state.  I had to make sure these two lines disappear before going forward to make the test successfully run. 
```
logging-mux-.*-deploy    Running
logging-mux-PREVIOUS_MUX_POD    Terminating
```